File: shared/src/types/training-plan.ts
export type RaceType = 'A' | 'B' | 'C';

export type SessionType =
  | 'easy_run'
  | 'long_run'
  | 'tempo'
  | 'intervals'
  | 'recovery'
  | 'race'
  | 'rest';

export interface TrainingPlan {
  id: string;
  userId: string;
  name: string;
  goalEvent: string;
  goalDate: Date;
  createdAt: Date;
  isActive: boolean;
  runsPerWeek: number;
  easyRunDay?: number | null;
  longRunDay?: number | null;
  intervalRunDay?: number | null;
}

export interface Race {
  id: string;
  planId: string;
  name: string;
  date: Date;
  distanceKm: number;
  type: RaceType;
  location?: string;
}

export interface TrainingWeek {
  id: string;
  planId: string;
  weekNumber: number;
  startDate: Date;
  focus?: string;
  weeklyVolumeKm: number;
  isTaperWeek: boolean;
  isCutbackWeek: boolean;
}

export interface TrainingSession {
  id: string;
  weekId: string;
  date: Date;
  sessionType: SessionType;
  description?: string;
  plannedDistanceKm?: number;
  plannedDurationMin?: number;
  completed: boolean;
  stravaActivityId?: string;
}

export interface CreateTrainingPlanRequest {
  name: string;
  goalEvent: string;
  goalDate: string; // ISO date string
  currentWeeklyVolumeKm: number;
  runsPerWeek?: number;
  easyRunDay?: number;
  longRunDay?: number;
  intervalRunDay?: number;
}

export interface PlanSummary extends TrainingPlan {
  totalWeeks: number;
  completedSessions: number;
  totalSessions: number;
  races: Race[];
}

File: backend/Models/TrainingPlan.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace MyRunna.Api.Models;

[Table("training_plans")]
public class TrainingPlan
{
    [Column("id")] public Guid Id { get; set; } = Guid.NewGuid();
    [Column("user_id")] public Guid UserId { get; set; }
    [Column("name")] public string Name { get; set; } = string.Empty;
    [Column("goal_event")] public string GoalEvent { get; set; } = string.Empty;
    [Column("goal_date")] public DateOnly GoalDate { get; set; }
    [Column("is_active")] public bool IsActive { get; set; } = true;
    [Column("current_weekly_volume_km")] public double CurrentWeeklyVolumeKm { get; set; }
    [Column("runs_per_week")] public int RunsPerWeek { get; set; } = 3;
    [Column("easy_run_day")] public int? EasyRunDay { get; set; }
    [Column("long_run_day")] public int? LongRunDay { get; set; }
    [Column("interval_run_day")] public int? IntervalRunDay { get; set; }
    [Column("created_at")] public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    [Column("updated_at")] public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public User User { get; set; } = null!;
    public ICollection<TrainingWeek> Weeks { get; set; } = [];
    public ICollection<Race> Races { get; set; } = [];
}
